flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/sink/ExactlyOnceKafkaWriter.java

```diff
@@ -277,6 +286,13 @@ private void abortLingeringTransactions(
         TransactionAbortStrategyContextImpl context =
                 getTransactionAbortStrategyContext(startCheckpointId, prefixesToAbort);
         transactionAbortStrategy.abortTransactions(context);
+        Collection<String> unsafeTransactionalIds = context.getUnsafeTransactionalIds();
+        if (!unsafeTransactionalIds.isEmpty()) {
+            namingContext.setUnsafeTransactionalIds(unsafeTransactionalIds);
+            LOG.info(
+                    "Unsafe transactional ids {} detected. They will be skipped until first checkpoint.",
```

Comment:
I do not fully understand the comment. 

I guess this relys on the fact that the committer during recovery tries to commit all outstanding transactions so we do not need to wait for `notifyCheckpointComplete`? My concern is that transactions are only committed on `notifyCheckpointComplete` that might not be aligned with the general checkpointing cycle.

I also find `skipped` without context hard to understand. Maybe something like `reused` would be easier.